experiments/compositions/expanders/helm-expander/main.go

```diff
@@ -0,0 +1,401 @@
+		content := []byte{}
+		if len(template.Content.Raw) != 0 {
+			yamlContent, err := yaml.JSONToYAML(template.Content.Raw)
+			if err != nil {
+				return fmt.Errorf("failed to marshall crds/%s file to yaml: %w", template.FileName, err)
+			}
+			content = yamlContent
+		} else if template.Template != "" {
+			content = []byte(template.Template)
+		}
+		err = atomicfile.WriteFile(filepath.Join(e.path, "templates", template.FileName), content, 0644)
+		if err != nil {
+			return fmt.Errorf("failed to write templates/%s file: %w", template.FileName, err)
+		}
+	}
+
+	valuesObj := map[string]interface{}{}
+
+	// Write Facade as values ?
+	if e.context != nil {
+		spec, found, err := unstructured.NestedMap(e.context.Object, "spec")
+		if err != nil {
+			return fmt.Errorf("failed to get .spec from Context: %w", err)
+		}
+		if !found {
+			return fmt.Errorf("Missing .spec from Context")
+		}
+		valuesObj["context"] = spec
+	}
+	if e.fetchedValues != nil {
+		valuesObj["fetched"] = *e.fetchedValues
+	}
+	if e.facade != nil {
+		spec, found, err := unstructured.NestedMap(e.facade.Object, "spec")
+		if err != nil {
+			return fmt.Errorf("failed to get .spec from Facade: %w", err)
+		}
+		if !found {
+			return fmt.Errorf("Missing .spec from Facade")
+		}
+
+		valuesObj["spec"] = spec
+	}
+	valuesBytes, err := json.Marshal(valuesObj)
+	if err != nil {
+		return fmt.Errorf("Unable to marshall values object")
+	}
+	err = atomicfile.WriteFile(filepath.Join(e.path, "facade-values.yaml"), valuesBytes, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write facade-values.yaml file: %w", err)
+	}
+	return nil
+}
+
+func (e *Expander) Validate(result *pb.ValidateResult) (*pb.ValidateResult, error) {
+	// https://helm.sh/docs/helm/helm_lint/
+	// Usage:
+	//  helm lint PATH [flags]
+	//
+	args := []string{"lint", e.path}
+	if e.facade != nil {
+		facadeFile := filepath.Join(e.path, "facade-values.yaml")
+		args = append(args, "-f", facadeFile)
+	}
+
+	op, err := exec.Command("helm", args...).CombinedOutput()
+	if err != nil {
+		result.Error.Message = fmt.Sprintf("failed validating template:\n %s", string(op))
+		result.Status = pb.Status_VALIDATE_FAILED
+		log.Print(result.Error.Message)
+		return result, nil
+	}
+
+	//log.Printf("Result:\n  type: %s\n  error: %s\n  status: %s\n", result.Type, result.Errors, result.Status)
+	return result, nil
+}
+
+func (e *Expander) Evaluate(result *pb.EvaluateResult) (*pb.EvaluateResult, error) {
+	// https://helm.sh/docs/helm/helm_template/
+	// Usage:
+	//  helm template [NAME] [CHART] [flags]
+	//
+	name := "unknown"
+	args := []string{"template", name, e.path}
+	if e.facade != nil {
+		name = e.facade.GetName()
+		facadeFile := filepath.Join(e.path, "facade-values.yaml")
+		args = append(args, "-f", facadeFile)
+	}
+
+	//args = append(args, "--output-dir", filepath.Join(e.path, "rendered"))
+
+	log.Printf("running: helm %s", strings.Join(args, " "))
+	op, err := exec.Command("helm", args...).CombinedOutput()
```

Comment:
Nit: It might be more robust to not combine stdout and stderr (although this assumes helm knows when to write to each...)